File: src/main/java/no/fintlabs/instance/gateway/model/oppfolgingstjenesten/IncomingInstance.java
package no.fintlabs.instance.gateway.model.oppfolgingstjenesten;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

import javax.validation.constraints.NotNull;


@Getter
@Jacksonized
@EqualsAndHashCode
@Builder
public class IncomingInstance {

    @NotNull
    private final String instansId;
    private final String dokumenttype;
    private final Personalia personalia;
    private final Kontaktinformasjon kontaktinformasjon;
    private final Inntaksadresse inntaksadresse;
    private final Dokument dokument;
}

File: src/main/kotlin/no/fintlabs/instance/gateway/model/vigo/IncomingInstance.kt
package no.fintlabs.instance.gateway.model.vigo

import com.fasterxml.jackson.annotation.JsonInclude
import javax.validation.constraints.NotNull

@JsonInclude(JsonInclude.Include.NON_NULL)
data class IncomingInstance(
    @field:NotNull
    val instansId: String,
    val dokumenttype: String? = null,
    val personalia: Personalia? = null,
    val kontaktinformasjon: Kontaktinformasjon? = null,
    val inntaksadresse: Inntaksadresse? = null,
    val dokument: Dokument? = null,
    val tilleggsinformasjon: Tilleggsinformasjon? = null
)
